### DISCUSSION TITLE
How to get all the nft id from full_node ?

### DISCUSSION THREAD

**Discussion by LCG22** [Chia-Network/chia-blockchain#17739] [Support]
I want to do it

**Comment by LCG22** [comment]
When I query the NFT IDs from the coin_record of the full node, I use the code:

coin_records = session.query(CoinRecord).filter(CoinRecord.puzzle_hash == LAUNCHER_PUZZLE_HASH).limit(1000).all()

The return includes some NFT IDs and others that are not NFT IDs. When I use the wallet to query these IDs to get NFT info, it shows the result:
"Failed to get NFT info: {'error': 'The coin is not a NFT.', 'success': False}"

So, how can I get only NFT IDs from coin_record?

**Reply by BrandtH22** [reply]
Hey, there is no direct method of querying all nft ids from the block records. One could use the singleton launcher puzzle (it sounds like this is what you are doing) then determine which singletons are nfts from there. Or one could use one of the third party services that works to help with more advanced block record requests like this:
https://mojonode.com/explorer?q=select%20date_trunc(%27week%27,%20spent_at)%20as%20%22datetime%22,%0A%20%20%20count(*)%20filter%20(where%20puzzle-%3E%3E%27m%27%20in%20(%2724e044101e57b3d8c908b8a38ad57848afd29d3eecc439dba45f4412df4954fd%27,%20%277faa3253bfddd1e0decb0906b2dc6247bbc4cf608f58345d173adb63e8b47c9f%27))%0A%20%20%20%20%20%20%20%20%20%20as%0A%20%20%20%20%20%20%20%20%20%20%20%20%22singleton%22,%0A%20%20%20count(*)%20filter%20(where%20puzzle-%3E%3E%27m%27%20=%20%27adb656e0211e2ab4f42069a4c5efc80dc907e7062be08bf1628c8e5b6d94d25b%27)%20as%20%22p2%20singleton%22,%0A%20%20%20count(*)%20filter%20(where%20puzzle-%3E%3E%27m%27%20=%20%2737bef360ee858133b69d595a906dc45d01af50379dad515eb9518abb7c1d2a7a%27)%20as%20%22CAT2%22,%0A%20%20%20count(*)%20filter%20(where%20puzzle-%3E%3E%27m%27%20=%20%2772dec062874cd4d3aab892a0906688a1ae412b0109982e1797a170add88bdcdc%27)%20as%20%22CAT1%22,%0A%20%20%20count(*)%20filter%20(where%20puzzle-%3E%3E%27m%27%20=%20%277a32d2d9571d3436791c0ad3d7fcfdb9c43ace2b0f0ff13f98d29f0cc093f445%27)%20as%20%22nft%20did%20launcher%22,%0A%20%20%20--count(*)%20filter%20(where%20puzzle-%3E%27a%27-%3E%27%27%20=%20%277a32d2d9571d3436791c0ad3d7fcfdb9c43ace2b0f0ff13f98d29f0cc093f445%27)%20as%20%22nft%20did%20launcher%22,%0A%20%20count(*)%20filter%20(where%20puzzle_hash%20in%20(%27%5Cxcfbfdeed5c4ca2de3d0bf520b9cb4bb7743a359bd2e6a188d19ce7dffc21d3e7%27,%20%0A%20%20%27%5Cxbae24162efbd568f89bc7a340798a6118df0189eb9e3f8697bcea27af99f8f79%27))%20as%20%22offer%22,%0A%20%20count(*)%20filter%20(where%20puzzle-%3E%27a%27-%3E1-%3E%27a%27-%3E%3E0%20=%20%27a04d9f57764f54a43e4030befb4d80026e870519aaa66334aef8304f5d0393c2%27%0A%20%20%20%20%20%20)%20as%20%22NFT1%22%0A%20%20%20from%20chia.coin_records%20where%20spent_at%20%3E%20%272022-11-01%27%0A%20%20group%20by%20%22datetime%22%20order%20by%20%22datetime%22

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
